pretyping/evarconv.ml

```diff
@@ -537,27 +592,91 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) ts env evd pbty
 		try 
 		  let termM' = Retyping.expand_projection env evd p c [] in
 		  let apprM', cstsM' = 
-		    whd_betaiota_deltazeta_for_iota_state
-		      (fst ts) env evd cstsM (termM',skM)
+                    whd_betaiota_deltazeta_for_iota_state flags.open_ts env evd cstsM (termM',skM)
 		  in
 		  let delta' i = 
-		    switch (evar_eqappr_x ts env i pbty) (apprF,cstsF) (apprM',cstsM') 
+                    switch (evar_eqappr_x flags env i pbty) (apprF,cstsF) (apprM',cstsM')
 		  in
-		    fun i -> ise_try i [f1; consume apprF apprM'; delta']
+                  fun i -> ise_try i [miller on_left ev apprF apprM';
+                                   consume on_left apprF apprM'; delta']
 		with Retyping.RetypeError _ ->
 		(* Happens thanks to w_unify building ill-typed terms *) 
 		  default
 	      in f evd
 	  end
       | _ -> default evd
   in
-  let flex_rigid on_left ev (termF, skF as apprF) (termR, skR as apprR) =
+  let rec first_order flexrigid env i t1 t2 sk1 sk2 =
+    (* Try first-order unification *)
+    match ise_stack2 false env i (evar_conv_x flags) sk1 sk2 with
```

Comment:
The thing is that zipping is expensive I guess, so we don't want to factor here. Note that this is just a refactorization of the original code (previously from L596 on). The changes to evar_conv itself is this refactoring and the passing of a record of flags instead of a tuple. Indeed I had generalized this first_order by a flexrigid flag at some point but it is not used in the end, I will clean that up.

Reply:
I see, so I'm fine with not factoring this code.

Reply:
I removed the flexrigid as it is unused